pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -530,9 +575,26 @@ func (fc *fluxForCluster) setupRepository(ctx context.Context) error {
 	return nil
 }
 
-// clone attempts to describe the provided remote repository.
-// if the repo exists, it will clone it.
-func (fc *fluxForCluster) clone(ctx context.Context) (*git.Repository, error) {
+func (fc *fluxForCluster) clone(ctx context.Context) error {
+	logger.V(3).Info("Cloning remote repository")
+	err := fc.FluxAddonClient.retrier.Retry(func() error {
+		return fc.gitTools.Client.Clone(ctx)
+	})
+	if err != nil {
+		return err
+	}
+
+	logger.V(3).Info("Creating a new branch")
+	err = fc.gitTools.Client.Branch(fc.branch())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// cloneIfExists attempts to describe the provided remote repository via the Git Provider.
```

Comment:
Won't ls-remote check for this too? Or is this specifically calling GitHub api? If we are separating these functions based on provider then it shows a bigger selling point for having providers separate.

Reply:
see above comment and https://github.com/aws/eks-anywhere/issues/1993